core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -320,7 +322,7 @@ class BlockManagerMasterEndpoint(
     // Find all shuffle blocks on executors that are no longer running
     val blocksToDeleteByShuffleService =
       new mutable.HashMap[BlockManagerId, mutable.HashSet[BlockId]]
-    if (externalBlockStoreClient.isDefined) {
+    if (externalShuffleServiceRemoveShuffleEnabled) {
```

Comment:
Can we keep the existing `externalBlockStoreClient.isDefined` condition behavior independently from this new configuration?

Reply:
I'm not really sure what you mean. That variable is `externalBlockStoreClient.isDefined && conf.get(config.SHUFFLE_SERVICE_REMOVE_SHUFFLE_ENABLED)`